### ISSUE TITLE
Cannot omit auth provider on next branch

### ISSUE THREAD

**Issue by jeiea** [marmelab/react-admin#5201]
**What you were expecting:**
Minimal example should cause no error.

**What happened instead:**

<details>
<summary>
`TypeError: Cannot read property 'getIdentity' of undefined.`
</summary>

```plaintext
Uncaught TypeError: Cannot read property 'getIdentity' of undefined
    at useGetIdentity.js:81
    at commitHookEffectListMount (react-dom.development.js:19731)
    at commitPassiveHookEffects (react-dom.development.js:19769)
    at HTMLUnknownElement.callCallback (react-dom.development.js:188)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:237)
    at invokeGuardedCallback (react-dom.development.js:292)
    at flushPassiveEffectsImpl (react-dom.development.js:22853)
    at unstable_runWithPriority (scheduler.development.js:653)
    at runWithPriority$1 (react-dom.development.js:11039)
    at flushPassiveEffects (react-dom.development.js:22820)
    at performSyncWorkOnRoot (react-dom.development.js:21737)
    at react-dom.development.js:11089
    at unstable_runWithPriority (scheduler.development.js:653)
    at runWithPriority$1 (react-dom.development.js:11039)
    at flushSyncCallbackQueueImpl (react-dom.development.js:11084)
    at flushSyncCallbackQueue (react-dom.development.js:11072)
    at unbatchedUpdates (react-dom.development.js:21909)
    at legacyRenderSubtreeIntoContainer (react-dom.development.js:24757)
    at Object.render (react-dom.development.js:24840)
    at Module../src/index.tsx (index.tsx:7)
    at __webpack_require__ (bootstrap:784)
    at fn (bootstrap:150)
    at Object.1 (index.tsx:9)
    at __webpack_require__ (bootstrap:784)
    at checkDeferredModules (bootstrap:45)
    at Array.webpackJsonpCallback [as push] (bootstrap:32)
    at main.chunk.js:1
```
</details>

**Steps to reproduce:**

1. Checkout react-admin next branch,
2. Run `make build`
3. Link example to next branch react-admin package.
4. Run `yarn start` on example.

**Related code:**
<details>
<summary>Example code</summary>

```javascript
import React from 'react';
import { Admin, Resource, List, Datagrid, TextField } from 'react-admin';

const PostList = props => (
  <List {...props}>
    <Datagrid>
      <TextField source="id" />
      <TextField source="title" />
      <TextField source="body" />
    </Datagrid>
  </List>
);

const data = [
  { id: 1, title: 'item 1', body: 'blah blah' },
  { id: 2, title: 'item 2', body: 'blah blah 2' },
  { id: 3, title: 'item 3', body: 'blah blah 3' },
];

const dataProvider = {
  getList: (resource, params) => Promise.resolve({ data, total: data.length }),
  getOne: (resource, params) => Promise.resolve({ data, total: data.length }),
  getMany: (resource, params) => Promise.resolve({ data, total: data.length }),
  getManyReference: (resource, params) =>
    Promise.resolve({ data, total: data.length }),
  create: (resource, params) => Promise.resolve({}),
  update: (resource, params) => Promise.resolve({}),
  updateMany: (resource, params) => Promise.resolve({}),
  delete: (resource, params) => Promise.resolve({}),
  deleteMany: (resource, params) => Promise.resolve({}),
};

export default () => (
  <Admin dataProvider={dataProvider}>
    <Resource name="posts" list={PostList} />
  </Admin>
);
```
</details>

**Other information:**

It seems `useGetIdentity` introduced recently.
```typescript
const useGetIdentity = () => {
    const [state, setState] = useSafeSetState<State>({
        loading: true,
        loaded: false,
    });
    const authProvider = useAuthProvider();
    useEffect(() => {
        if (typeof authProvider.getIdentity === 'function') {
            const callAuthProvider = async () => {
                try {
```
I think auth provider check is missing here.

**Environment**

* React-admin version: commit SHA `d58084418`
* React version: latest stable (16.x)
* Browser: Chrome

**Comment by djhi** [comment]
Thanks for reporting.

**Comment by djhi** [comment]
Fixed by #5201
